app/features/project/project_controller.rb

```diff
@@ -70,77 +72,87 @@ class ProjectController < AuthenticatedControllerBase
       erb(:new_project, locals: locals, layout: FastlaneCI.default_layout)
     end
 
-    get "#{HOME}/add/*" do |repo_name|
+    # This is an utility endpoint from where we can retrieve lane information through the front-end using basic JS.
+    # This will be reviewed in the future when we have a proper front-end architecture.
+    get "#{HOME}/lanes/*/*/*" do |org, repo_name, branch|
+      content_type :json
+
       provider_credential = check_and_get_provider_credential(type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github])
 
       github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
-      selected_repo = github_service.repos.select { |repo| repo_name == repo.name }.first
+      selected_repo = github_service.repos.detect { |repo| org + "/" + repo_name == repo.full_name }
 
-      # We need to check whether we can checkout the project without issues.
-      # So a new project is created with default settings so we can fetch it.
       repo_config = GitRepoConfig.from_octokit_repo!(repo: selected_repo)
 
       dir = Dir.mktmpdir
-      repo = GitRepo.new(
-        git_config: repo_config,
-        provider_credential: provider_credential,
-        local_folder: dir,
-        async_start: false
+      repo = FastlaneCI::GitRepo.new(git_config: repo_config,
+                                    local_folder: dir,
+                                    provider_credential: provider_credential,
+                                    async_start: false)
+
+
+      fastfile = FastlaneCI::FastfilePeeker.peek(
+        git_repo: repo,
+        branch: branch
       )
 
-      # TODO: This should be refactored in some kind of FastlaneUtils` class.`
-      # We have synchronously cloned the repo, now we need to get the lanes.
-      repo_path = repo.local_folder
-      # First assume the fastlane directory and its file is in the root of the project
-      fastfiles = Dir[File.join(repo_path, "fastlane/Fastfile")]
-      # If not, it might be in a subfolder
-      fastfiles = Dir[File.join(repo_path, "**/fastlane/Fastfile")] if fastfiles.count == 0
-
-      if fastfiles.count > 1
-        logger.error("Ugh, multiple Fastfiles found, we're gonna have to build a selection in the future")
-        # for now, just take the first one
+      fastfile_config = {}
+
+      fastfile.tree.each_key do |key|
```

Comment:
I'm not sure on why we need to do this, maybe add a few lines of comment explaining it?

Reply:
Yeah just added everything super fast in this and didn't add comments, I'll do it in a follow up